PublicDashboards: disable annotations

evictorero opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

-->

**What this PR does / why we need it**:
Annotations require special permissions to be retrieved, we are removing them from public dashboards v1.
This PR only removes retrieving annotations for public dashboards.

Disabling annotations for public dashboards.

**Which issue(s) this PR fixes**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes https://github.com/grafana/grafana-enterprise-partnerships-team/issues/222

evictorero committed:
PublicDashboards: disable annotations

jalevin reviewed (changes requested):
This looks great!

Only feedback is that I think it would be worthwhile to add a test for when isPublic: true.

torkelo reviewed (changes requested):
This sadly is not enough. It only addresses the default auto annotation query that every dashboard has that fetches annotations stored in Grafanas main db . 

But users can add annotation queries that target other datasources.

I can try to come up with a good place to add this condition to tomorrow

evictorero committed:
Revert "PublicDashboards: disable annotations"

This reverts commit 6d96644b93f811ed18c80d8f80d348100a55dd97.

evictorero committed:
PublicDashbaords: disabling annotations worker for public dashboards

evictorero commented:
@torkelo thanks for taking a look at this. I reverted the last commit and added a check to the `canWork` function of the `AnnotationsWorker` to avoid fetching annotations from all datasources when the dashboard is pulic and also is being requested in _Public view mode_.  Let me know what do you think.

@jalevin removed all code and added some unit tests for this new approach

evictorero committed:
changing uid by accessToken

evictorero committed:
remove unused properties used for test and update betterer results
